### PR TITLE
Adjust nightly_cleanup to accommodate multitenant

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

as @lomky pointed out, some of the stuff in nightly_cleanup depends on fund-level configs, so we gotta scope some of these to fund. In general, if it's got a fund_id on the table, we do it in the fund loop now.

Others of these we can do all at once, like clinic coordinates updates, clearing out old events, or trimming db sessions, though, so those can stay out o the loop.

This pull request makes the following changes:
* do fund-specific cleanup subtasks in the context of a fund

no views

It relates to the following issue #s: 
* Bumps #1817 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: lib/tasks/nightly_cleanup.rake [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
-  puts "#{Time.now} -- cleared all recently reached patients from call lists"
+  Rake::Task['db:sessions:trim'].invoke
+  puts "#{Time.now} -- removed old sessions"
 
-  User.disable_inactive_users
-  puts "#{Time.now} -- locked accounts of users who have not logged in since #{User::TIME_BEFORE_DISABLED_BY_FUND.ago}"
+  Fund.all.each do |fund|
+    ActsAsTenant.with_tenant(fund) do
+      User.all.each { |user| user.clean_call_list_between_shifts }
+      puts "#{Time.now} -- cleared all recently reached patients from call lists"
```
Comment (lomky) [inline lib/tasks/nightly_cleanup.rake:10]: Whatcha think about adding the fund to the logging msg?

```suggestion
      puts "#{Time.now} -- cleared all recently reached patients from call lists for fund #{fund}"
```

repeat for each?

File: lib/tasks/nightly_cleanup.rake [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
+      Event.destroy_old_events
+      puts "#{Time.now} -- destroyed old events"
 
-  Rake::Task['db:sessions:trim'].invoke
-  puts "#{Time.now} -- removed old sessions"
+      if Time.zone.now.monday?
+        # Run these events weekly
+        Clinic.update_all_coordinates
+        puts "#{Time.now} -- refreshed coordinates on all clinics"
+      end
```
Comment (lomky) [inline lib/tasks/nightly_cleanup.rake:25]: Description says this one can be fund agnostic - but its per fund here?

**Comment by colinxfleming** [comment]
wow was my brain paste when I wrote this one. bumping now, thank you @lomky 😅

**Review by lomky — APPROVED** [review]
🎉

**Pull request merged**